domain/user/service/service.go

```diff
@@ -188,35 +258,53 @@ func (s *Service) AddUserWithPassword(ctx context.Context, user user.User, passw
 // - usererrors.AlreadyExists: If a user with the supplied name already exists.
 // - usererrors.UserCreatorNotFound: If a creator has been supplied for the user
 // and the creator does not exist.
-func (s *Service) AddUserWithActivationKey(ctx context.Context, user user.User) ([]byte, error) {
-	if err := ValidateUsername(user.Name); err != nil {
-		return nil, fmt.Errorf("username %q with activation key: %w", user.Name, err)
+func (s *Service) AddUserWithActivationKey(ctx context.Context, usr user.User, creatorUUID user.UUID) ([]byte, error) {
+	// Validate user name and creator name
+	if err := ValidateUsername(usr.Name); err != nil {
+		return nil, fmt.Errorf("username %q with activation key: %w", usr.Name, err)
+	}
+
+	// Validate creator UUID
+	if err := creatorUUID.Validate(); err != nil {
+		return nil, fmt.Errorf("creator uuid %q with activation key: %w", creatorUUID, err)
 	}
 
+	// Generate an activation key for the user.
 	activationKey, err := generateActivationKey()
 	if err != nil {
-		return nil, fmt.Errorf("generating activation key for user %q: %w", user.Name, err)
+		return nil, fmt.Errorf("generating activation key for user %q: %w", usr.Name, err)
+	}
+
+	// Get creator UUID for creatorUsername
+	creatorUser, err := s.GetUser(ctx, creatorUUID)
```

Comment:
Small nit. I think move this block up above the creation of the activation key. Let's make sure all of our pre conditions are satisfied before doing any lifting.